typed-racket-lib/typed-racket/typecheck/tc-toplevel.rkt

```diff
@@ -400,6 +400,8 @@
           [else (int-err "Two conflicting definitions: ~a ~a" def other-def)]))
       (dict-update h (binding-name def) merge-def-bindings #f)))
   (do-time "computed def-tbl")
+  ;; check that all parsed apps are sensible:
+  (check-registered-apps!)
```

Comment:
I added another check after pass 2. At a glance that seems like it should cover all the Apps we could run into while type checking a file.